language/diem-framework/modules/DiemSystem.move

```diff
@@ -290,17 +295,20 @@ module DiemFramework::DiemSystem {
         let is_validator_info_updated = update_ith_validator_info_(&mut diem_system_config.validators, to_update_index);
         if (is_validator_info_updated) {
             let validator_info = Vector::borrow_mut(&mut diem_system_config.validators, to_update_index);
-            assert(DiemTimestamp::now_microseconds() >
-                   validator_info.last_config_update_time + FIVE_MINUTES,
-                   ECONFIG_UPDATE_RATE_LIMITED);
+            assert(
+                validator_info.last_config_update_time <= MAX_U64 - FIVE_MINUTES,
+                Errors::limit_exceeded(ECONFIG_UPDATE_TIME_OVERFLOWS)
+            );
+            assert(
+                DiemTimestamp::now_microseconds() > validator_info.last_config_update_time + FIVE_MINUTES,
+                Errors::invalid_state(ECONFIG_UPDATE_RATE_LIMITED)
```

Comment:
Is invalid state the appropriate category here?

Reply:
LIMIT_EXCEEDED might be better